From 3f9fa9c39267193a9fd27e8de965fa9eb6805f71 Mon Sep 17 00:00:00 2001
From: Nikolay Aleksandrov <nikolay@redhat.com>
Date: Fri, 5 Sep 2014 11:03:39 -0400
Subject: [ethernet] sfc: Fix transposed ptp_{under, over}size_sync_windows statistics

Message-id: <1409915048-26438-4-git-send-email-nikolay@redhat.com>
Patchwork-id: 91412
O-Subject: [RHEL7.1 PATCH BZ 1110888 03/32] sfc: Fix transposed ptp_{under, over}size_sync_windows statistics
Bugzilla: 1110888
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Ben Hutchings <bhutchings@solarflare.com>

Somehow I transposed these two while bringing the original statistics
support upstream.

Fixes: 99691c4ac112 ('sfc: Add PTP counters to ethtool stats')
Signed-off-by: Ben Hutchings <bhutchings@solarflare.com>
Signed-off-by: Shradha Shah <sshah@solarflare.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 13c92e82e821d87e388a73ffeddeb62d753dab96)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/sfc/ptp.c b/drivers/net/ethernet/sfc/ptp.c
index 7aa0708..eb75fbd 100644
--- a/drivers/net/ethernet/sfc/ptp.c
+++ b/drivers/net/ethernet/sfc/ptp.c
@@ -753,9 +753,9 @@ efx_ptp_process_times(struct efx_nic *efx, MCDI_DECLARE_STRUCT_PTR(synch_buf),
   if (window < SYNCHRONISATION_GRANULARITY_NS) {
    ++ptp->invalid_sync_windows;
   } else if (corrected >= MAX_SYNCHRONISATION_NS) {
-   ++ptp->undersize_sync_windows;
-  } else if (corrected < ptp->min_synchronisation_ns) {
    ++ptp->oversize_sync_windows;
+  } else if (corrected < ptp->min_synchronisation_ns) {
+   ++ptp->undersize_sync_windows;
   } else {
    ngood++;
    last_good = i;
-- 
1.7.1